File: kurs programowania/lista3/cpp/Szesciokat.cpp
#include "Szesciokat.hpp"

#include <iostream>
#include <math.h>

Szesciokat::Szesciokat(int a)
{
    if (a < 0)
        throw std::runtime_error("illegal_args");

    this->a = a;
}

double Szesciokat::pole()
{
    return 3 * a * a * sqrt(3) / 2;
}

double Szesciokat::obwod()
{
    return 6 * a;
}

File: kurs programowania/lista3/zadanie1/Szesciokat.java
public class Szesciokat extends Figura {

    public int bok;

    public Szesciokat(int bok) {

        if (bok < 0)
            throw new IllegalArgumentException();

        this.bok = bok;
    }

    public double pole() {

        return 3 * bok * bok * Math.sqrt(3) / 2;
    }

    public double obwod() {

        return 6 * bok;
    }
}